src/website/webview/views.py

```diff
@@ -409,31 +411,20 @@ def get_object(self, queryset: QuerySet | None = None) -> NixpkgsIssue:
         issue = cast(
             NixpkgsIssue, get_object_or_404(self.model, code=self.kwargs.get("code"))
         )
-        derivations = issue.derivations.all()  # type: ignore
-        for drv in derivations:
-            result = self.get_cves_for_derivation(drv)
-            drv.known_cves = result
-
-        issue.derivations_with_cves = derivations  # type: ignore
-
+        issue.cached_payload = CachedNixpkgsIssuePayload(**issue.cached.payload)  # type: ignore
         return issue
 
-    def get_cves_for_derivation(self, drv: Any) -> QuerySet | None:
-        known_vulnerabilities = drv.metadata.known_vulnerabilities
-        if not known_vulnerabilities:
-            return None
-        cves = [s for s in known_vulnerabilities if self.pattern.match(s)]
-        existing_cves = Container.objects.filter(cve__cve_id__in=cves)
-        return existing_cves or None
-
 
 class NixpkgsIssueListView(ListView):
     template_name = "issue_list.html"
     model = NixpkgsIssue
     paginate_by = 10
 
     def get_queryset(self) -> BaseManager[NixpkgsIssue]:
-        return NixpkgsIssue.objects.all().order_by("-created")
+        issues = NixpkgsIssue.objects.all().order_by("-created")
+        for issue in issues:
+            issue.cached_payload = CachedNixpkgsIssuePayload(**issue.cached.payload)  # type: ignore
+        return issues
```

Comment:
Why not

```suggestion
        return [CachedNixpkgsIssuePayload(**issue.cached.payload) for issue in issues]
```

(and change the return type to the pydantic schema)?

That would avoid a lot of visual noise in the template

Reply:
The view uses the db key for issues as a title and it is not part of the pydantic schema. It used to be but it turns out it is only present when manually generating cached issues, not when it happens via the hook.

Would you like to return pairs of pk/cached issue in the view?